src/librustdoc/config.rs

```diff
@@ -372,15 +372,16 @@ impl Options {
                                                 .iter()
                                                 .map(|s| (PathBuf::from(&s), s.to_owned())) {
                 if !theme_file.is_file() {
-                    diag.struct_err("option --themes arguments must all be files").emit();
+                    diag.struct_err(&format!("invalid file: \"{}\"", theme_s))
+                        .help("option --themes arguments must all be files")
+                        .emit();
                     return Err(1);
                 }
                 let (success, ret) = theme::test_theme_against(&theme_file, &paths, &diag);
                 if !success || !ret.is_empty() {
-                    diag.struct_err(&format!("invalid theme: \"{}\"", theme_s))
+                    diag.struct_warn(&format!("invalid theme: \"{}\"", theme_s))
```

Comment:
I feel like we may want to have different messages for `!success` (which is given when there is an error loading the file) and `!ret.is_empty()` (which is given when there are missing CSS rules in the given theme). For one, if there was a file error, telling people to use `--theme-checker` isn't going to help.

How about something like this?

```rust
if !success {
    diag.struct_warn(&format!("error loading theme file: \"{}\"", theme_s)).emit();
} else if !ret.is_empty() {
    diag.struct_warn(&format!("theme file \"{}\" is missing CSS rules from the default theme", theme_s))
        .warn("the theme may appear incorrect when loaded")
        .help(&format!("to see what rules are missing, call `rustdoc --theme-checker \"{}\"`", theme_s))
        .emit();
}
```

It might be worth still exiting if the file couldn't be loaded? If we wait all the way until HTML generation to bail (which will happen if the file is still unavailable when the HTML is being rendered), then we've wasted the entire compilation process on something we've already checked right here.

Reply:
I kept the return in case of file load error.